libr/bin/format/elf/elf.c

```diff
@@ -4021,3 +4013,11 @@ char *Elf_(r_bin_elf_compiler)(ELFOBJ *bin) {
 	free (buf);
 	return res;
 }
+
+bool Elf_(r_bin_elf_is_executable)(ELFOBJ *bin) {
+	switch (bin->ehdr.e_type) {
```

Comment:
probably this will be shorter and more readable (and wont need to make the compiler decide which cf construction to use):

```c
const int t = bin->ehdr.e_type;
return (t == ET_EXEC || t == ET_DYN);
```